metabase/metabase
# fix(sdk): emit typescript files in the embedding cli when in a typescript project

Closes https://github.com/metabase/metabase/issues/50094

### Description

We should emit TypeScript files when in a TypeScript project, i.e. when the project contains a `tsconfig.json` file, or has `typescript` in the project's `devDependencies`.

### How to verify

- Go to a TypeScript project. You can create one with Vite.
- Run the CLI. I usually use `docker rm -f metabase-enterprise-embedding && yarn add file:../metabase/resources/embedding-sdk && npx @metabase/embedding-sdk-react start`
- It should emit `.ts` and `.tsx` files.
- For regular JavaScript projects, It should still emit `.js` and `.jsx` files.

### Demo

In TypeScript projects, the file extensions now become ts and tsx.

![CleanShot 2567-11-20 at 21 29 08@2x](https://github.com/user-attachments/assets/90fda4ac-c6bd-45d2-bed0-0f271cdac297)
